qcodes/dataset/plotting.py

```diff
@@ -269,6 +288,110 @@ def plot_by_id(run_id: int,
     return axes, new_colorbars
 
 
+def _complex_to_real_preparser(alldata: NamedData,
+                               conversion: str,
+                               degrees: bool=False) -> NamedData:
+    """
+    Convert complex-valued parameters to two real-valued parameters, either
+    real and imaginary part or phase and magnitude part
+
+    Args:
+        alldata: the data to convert, should be the output of get_data_by_id
+        conversion: the conversion method, either "real_and_imag" or
+            "mag_and_phase"
+        degress: whether to return the phase in degrees. The default is to
+            return the phase in radians
+    """
+
+    if conversion not in ['real_and_imag', 'mag_and_phase']:
+        raise ValueError(f'Invalid conversion given. Received {conversion}, '
+                         'but can only accept "real_and_imag" or '
+                         '"mag_and_phase".')
+
+    newdata = []
+
+    # we build a new NamedData object from the given `alldata` input.
+    # Note that the length of `newdata` will be larger than that of `alldata`
+    # in the case of complex top-level parameters. This is the reason why we
+    # use two variables below,
+
+    for group in alldata:
+        new_group = []
+        new_groups: NamedData = [[], []]
+        for index, parameter in enumerate(group):
+            data: np.ndarray = parameter['data']
```

Comment:
`data` seems to be unused, right?
```suggestion
```

Reply:
I think we use it to check whether the data values are complex (next line).

Reply:
sono idioto :)